api/v1/server/handlers/v1/webhooks/create.go

```diff
@@ -0,0 +1,184 @@
+package webhooksv1
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/gen"
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/transformers/v1"
+	"github.com/hatchet-dev/hatchet/pkg/repository/postgres/dbsqlc"
+	"github.com/hatchet-dev/hatchet/pkg/repository/postgres/sqlchelpers"
+	v1 "github.com/hatchet-dev/hatchet/pkg/repository/v1"
+	"github.com/hatchet-dev/hatchet/pkg/repository/v1/sqlcv1"
+	"github.com/labstack/echo/v4"
+)
+
+func (w *V1WebhooksService) V1WebhookCreate(ctx echo.Context, request gen.V1WebhookCreateRequestObject) (gen.V1WebhookCreateResponseObject, error) {
+	tenant := ctx.Get("tenant").(*dbsqlc.Tenant)
+
+	canCreate, _, err := w.config.EntitlementRepository.TenantLimit().CanCreate(ctx.Request().Context(), dbsqlc.LimitResourceINCOMINGWEBHOOK, tenant.ID.String(), 1)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to check if webhook can be created: %w", err)
+	}
+
+	if !canCreate {
+		return gen.V1WebhookCreate400JSONResponse{
+			Errors: []gen.APIError{
+				{
+					Description: "incoming webhook limit reached",
+				},
+			},
+		}, nil
+	}
+
+	params, err := w.constructCreateOpts(tenant.ID.String(), *request.Body)
+	if err != nil {
+		return gen.V1WebhookCreate400JSONResponse{
+			Errors: []gen.APIError{
+				{
+					Description: fmt.Sprintf("failed to construct webhook create params: %v", err),
+				},
+			},
+		}, nil
+	}
+
+	webhook, err := w.config.V1.Webhooks().CreateWebhook(
+		ctx.Request().Context(),
+		tenant.ID.String(),
+		params,
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create webhook")
+	}
+
+	transformed := transformers.ToV1Webhook(webhook)
+
+	return gen.V1WebhookCreate200JSONResponse(transformed), nil
+}
+
+func extractAuthType(request gen.V1CreateWebhookRequest) (sqlcv1.V1IncomingWebhookAuthType, error) {
+	j, err := request.MarshalJSON()
+
+	if err != nil {
+		return "", fmt.Errorf("failed to get marshal request: %w", err)
+	}
+
+	parsedBody := make(map[string]interface{})
+	if err := json.Unmarshal(j, &parsedBody); err != nil {
+		return "", fmt.Errorf("failed to unmarshal request body: %w", err)
+	}
+
+	unparsedDiscriminator, ok := parsedBody["authType"]
+	if !ok {
+		return "", fmt.Errorf("authType field is missing in the request body")
+	}
+
+	discriminator := unparsedDiscriminator.(string)
+
+	authType := sqlcv1.V1IncomingWebhookAuthType(discriminator)
+	if authType == "" {
+		return "", fmt.Errorf("invalid auth type: %s", unparsedDiscriminator)
+	}
+
+	return authType, nil
+}
```

Comment:
This is a hack - I couldn't figure out how else to extract the discriminator because the discriminated union doesn't translate well (AFAICT) to Go types. would love some ideas for this one 😂 but it seems to work fine at least, just not type safe

Reply:
yeah I'm not sure, this seems reasonable - don't see anything helpful in https://github.com/oapi-codegen/oapi-codegen?tab=readme-ov-file#how-does-oapi-codegen-handle-anyof-allof-and-oneof